### PR TITLE
Logicalsc optimized

### PR DESCRIPTION
**Pull request by ldennis**
- fixed log deletion bug for logical redo based schema change
- optimized redo thread to not read log file if there is no concurrent writes against the table

### CODE REVIEW

File: schemachange/sc_records.c [hunk 1]
```diff
@@ -1527,9 +1526,18 @@ int convert_all_records(struct dbtable *from, struct dbtable *to,
         data.tagmap = NULL;
     }
 
+    if (s->logical_livesc) {
+        if (outrc == 0) {
+            s->sc_convert_done[MAXDTASTRIPE] = 1;
```
Comment (ldennis) [inline schemachange/sc_records.c]: set under the mutex

**Comment by markhannum** [comment]
Good to merge

**Comment by ldennis** [comment]
woah

**Comment by akshatsikarwar** [comment]
wow

**Pull request merged**